### DISCUSSION TITLE
How subscriptions are handled by broker ?

### DISCUSSION THREAD

**Discussion by aqibamir** [moscajs/aedes#795] [Q&A]
I'm trying to understand how aedes handels subscriptions. when aedes.on('subscribe', callback) is called does the default persistence automatically store or do I need to maintain subscriptions in code myself ?

Plus, I'm new to this and I assumed that aedes.subscribe is called every time a new subscription is called but i'm confused how to assign topic ?

**Comment by Gianluca-Casagrande-Stiga** [comment]
Hi, the messaging pub-sub is completely managed by aedes. 
The aedes event 'subscribe' is only emitted every time a mqtt client subscribes to a topic. If you want, you can not listen to it (I usually put a simple logger into the 'subscribe' listener).

**Reply by aqibamir** [reply]
hey, thanks for the answer. 
Btw, I'm trying to get/set value of a variable using mqtt. However, I've figured out setting the value, since I can do that by just publishing but I can not figure out how to retrieve the value everytime on demand. 
I'm thinking of subscribing everytime but I don't think that's the right way to do

**Reply by Gianluca-Casagrande-Stiga** [reply]
I think you should have one process with the broker (aedes) and many other with the clients. If you have doubts on how subscribe works, you should study it. I've used this tutorial for learning mqtt: https://info.hivemq.com/mqtt-essentials

**Comment by robertsLando** [comment]
> when aedes.on('subscribe', callback) is called does the default persistence automatically store or do I need to maintain subscriptions in code myself ?

That is just an event to inform you that a subscription has happen, all the underlying logic is handled by aedes (store subscription too but only if he needs to store it based on if session is clean or not and the sub QoS)

> Plus, I'm new to this and I assumed that aedes.subscribe is called every time a new subscription is called but i'm confused how to assign topic ?

aedes `subscribe` method is used internally to handle subscriptions when it receives a subscipption request from a client or you can also use it on your code in order to do subscriptions based on your needs. What I think you are missing here is that you need another library like mqttjs in order to connect clients to the broker, also like @Gianluca-Casagrande-Stiga  suggested I think you firstly need to read something about mqtt foundamentals before using aedes

**Reply by aqibamir** [reply]
Hey, thanks for the answer.  In my use case I have to send data from database whenever a device subscribes to a topic rather than the retained data. I'm trying to use the aedes.subscribe() but can't understand how to exactly use this. I believe the deliverfunc actually delivers the message but I can't figure out how I'd actually use it. Any help would be appreciated @robertsLando @Gianluca-Casagrande-Stiga

**Reply by robertsLando** [reply]
>  I have to send data from database whenever a device subscribes to a topic

https://github.com/moscajs/aedes/blob/main/docs/Aedes.md#event-subscribe
